kedro_azureml/config.py

```diff
@@ -79,6 +80,8 @@ class KedroAzureRunnerConfig(BaseModel):
   # Path to the directory in the Docker image to run the code from
   # Ignored when code_directory is set
   working_directory: /home/kedro_docker
+  # Use Azure ML native data passing instead of temporary storage
+  native_data_passing: false
```

Comment:
We've been discussing this internally and decided to name this feature differently.

In configuration, let's go with a section:
```yaml
pipeline_data_passing:
  enabled: false # disabled by default
  # this way we have a way to extend it in a future
```
and consequently, the let's name the dataset you've implemented: `AzureMLPipelineDataSet`. 

That way, the name is consistent in both config and DataSet; also - it does not introduce "native" keyword, which does not seem to occur in the Azure's docs related to data passing.